scarlet/psf.py

```diff
@@ -79,190 +107,88 @@ def f(X):
-    def kernel(self):
-        """Return the contents of the kernel parameter
-        """
-        return self.get_parameter(0)._data
-
-
-class PsfObservation(Observation):
-    def match(self, psfs, convolution="real"):
-        """Implement the observed PSFs as the difference kernel
 
-        This is different than `~scarlet.Observation.match`,
-        where the difference kernel that matches the model
-        PSF to the observed PSF is calculated and stored.
-        For a `PsfObservation` the input PSF, which
-        should be the observed PSF (see below) is stored
-        as the "difference kernel" while the actual
-        deconvolution (difference) kernel is calculated
-        by the model.
+class ImagePSF(PSF):
+    """Image PSF
 
-        Parameters
-        ----------
-        psfs: `~numpy.array`
-            The input PSF that is being convolved.
-            For deconvolution this is the observed PSF,
-            since the observed PSF is convolved with the
-            deconvolution kernel to match the model PSF.
-            For matching a model PSF to a wider observed
-            PSF use the `Observation` class,
-            which calculates the difference kernel much
-            more quickly using a DFT.
+    Creates a PSF model from a image of the centered PSF.
 
-        Returns
-        -------
-        self: `~scarlet.PsfObservation`
-            Return this object to allow for chaining.
-        """
-        self._diff_kernels = Fourier(psfs)
-        self.convolution = convolution
-        return self
+    Parameters
+    ----------
+    image: 2D or 3D array
+    """
 
-    def get_loss(self, model):
-        """get_loss
+    def __init__(self, image):
 
-        We override `scarlet.Observation.get_loss` since the lognorm
-        is the dominant term, which interferes with calculating
-        relative error.
-        """
-        model_ = self.render(model)
-        images_ = self.images
-        weights_ = self.weights
-        return np.sum(weights_ * (model_ - images_) ** 2) / 2
+        if len(image.shape) == 2:
+            shape = image.shape
+            image = image.reshape(1, *shape)
 
+        image = self.normalize(image)
+        image = self.prepare_param(image, "image")
+        super().__init__(image)
 
-class DeconvolvedObservation(Observation):
-    """Deconvolved image using a deconvolution kernel
+        origin = (0, -(image.shape[1] // 2), -(image.shape[2] // 2))
+        self.bbox = Box(image.shape, origin=origin)
 
-    This is the same as `~scarlet.Observation` except that the matching
-    algorithm uses a precalculated deconvolution kernel and the
-    images are deconolved using that kernel. A separate class exists
-    to make it less likely that the user executes this operation
-    more than once, which will have adverse affects on the
-    `images` property.
-    """
+    def get_model(self, *parameters, offset=None):
+        assert offset is None, "ImagePSF cannot be offset"
```

Comment:
But it can with FFT! Why not implement it?

Reply:
Because of the artifacts that arise, it isn't a good idea to shift an `ImagePSF`. That said I will implement it even if we generally should discourage model PSFs from images.